homeassistant/components/binary_sensor/rfxtrx.py

```diff
@@ -0,0 +1,235 @@
+from homeassistant.components import rfxtrx
+from homeassistant.util import slugify
+from homeassistant.util import dt as dt_util
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import event as evt
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.rfxtrx import (
+    ATTR_AUTOMATIC_ADD, ATTR_NAME, ATTR_OFF_DELAY, ATTR_FIREEVENT,
+    ATTR_DATA_BITS, CONF_DEVICES
+)
+from homeassistant.const import (
+    CONF_SENSOR_CLASS, CONF_COMMAND_ON, CONF_COMMAND_OFF
+)
+
+DEPENDENCIES = ["rfxtrx"]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required("platform"): rfxtrx.DOMAIN,
+    vol.Optional(CONF_DEVICES, default={}): vol.All(
+        dict, rfxtrx.valid_binary_sensor),
+    vol.Optional(ATTR_AUTOMATIC_ADD, default=False):  cv.boolean,
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup the Binary Sensor platform to rfxtrx."""
+    import RFXtrx as rfxtrxmod
+    sensors = []
+
+    for packet_id, entity in config['devices'].items():
+        event = rfxtrx.get_rfx_object(packet_id)
+        device_id = slugify(event.device.id_string.lower())
+
+        if device_id in rfxtrx.RFX_DEVICES:
+            continue
+
+        if entity[ATTR_DATA_BITS] is not None:
+            _LOGGER.info("Masked device id: %s",
+                         rfxtrx.get_pt2262_deviceid(device_id,
+                                                    entity[ATTR_DATA_BITS]))
+
+        _LOGGER.info("Add %s rfxtrx.binary_sensor (class %s)",
+                     entity[ATTR_NAME], entity[CONF_SENSOR_CLASS])
+
+        device = RfxtrxBinarySensor(event, entity[ATTR_NAME],
+                                    entity[CONF_SENSOR_CLASS],
+                                    entity[ATTR_FIREEVENT],
+                                    entity[ATTR_OFF_DELAY],
+                                    entity[ATTR_DATA_BITS],
+                                    entity[CONF_COMMAND_ON],
+                                    entity[CONF_COMMAND_OFF])
+
+        sensors.append(device)
+        rfxtrx.RFX_DEVICES[device_id] = device
+
+    add_devices_callback(sensors)
+
+    # pylint: disable=too-many-branches
+    def binary_sensor_update(event):
+        """Callback for control updates from the RFXtrx gateway."""
+        if not isinstance(event, rfxtrxmod.ControlEvent):
+            return
+
+        device_id = slugify(event.device.id_string.lower())
+
+        if device_id in rfxtrx.RFX_DEVICES:
+            sensor = rfxtrx.RFX_DEVICES[device_id]
+        else:
+            sensor = rfxtrx.get_pt2262_device(device_id)
+
+        if sensor is None:
+            # Add the entity if not exists and automatic_add is True
+            if config[ATTR_AUTOMATIC_ADD]:
+                poss_dev = rfxtrx.find_possible_pt2262_device(device_id)
+
+                if poss_dev is not None:
```

Comment:
I'd like to show this to the user in the web interface but I didn't have time to look for a way to do it properly. Is there a way to do so ?

I don't consider 'automatic_add' being a *permanent* option in the configuration files, as the binary_sensor will also intercept events from switches.